internal/controller/clusterdeployment_controller.go

```diff
@@ -271,6 +274,13 @@ func (r *ClusterDeploymentReconciler) updateCluster(ctx context.Context, cd *kcm
 			"namespace":  cred.Spec.IdentityRef.Namespace,
 		}
 
+		if r.EnableGlobalVals {
```

Comment:
Isn't this boolean parameter excessive? Can it be replaced with:
```
if r.GlobalRegistry != "" || r.GlobalK0sURL != ""
```
or something like this?